### PR TITLE
[Add] Global streams

### PR DESCRIPTION
**Pull request by siva2204**
- add notification, transaction, stockprice as global stream

### CODE REVIEW

File: lib/global_streams.dart [hunk 1]
```diff
@@ -1,60 +1,116 @@
...
+        stockResponse.stockList,
+        gameStateStream,
+        notificationStream,
+        transactionStream,
+        stockPriceStream,
+        subscriptionIds);
+  } catch (e) {
+    logger.e(e.toString());
+    throw Exception(e.toString());
   }
```
Comment (Sudhindra3000) [inline lib/global_streams.dart:122]: Why catch the exception and throw it again if we're not doing any changes

File: lib/grpc/subscription.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      }
+
+      logger.d(
+          '[$subscribeResponse.subscriptionId.dataStreamType}]: subscribe successful');
+
+      return subscribeResponse.subscriptionId;
+    } catch (e) {
+      logger.e(e);
+      throw failedToReachServerError;
+    }
```
Comment (Sudhindra3000) [inline lib/grpc/subscription.dart:43]: Remove try catch in these two function?
It'd be better for the caller to handle exception

File: lib/constants/exception.dart [hunk 3]
```diff
@@ -0,0 +1,3 @@
+final invalidDataStreamIdError = Exception('Invalid data stream id');
+final internalServerError = Exception('Internal server error');
+final failedToReachServerError = Exception('Unable reach the server');
```
Comment (Sudhindra3000) [inline lib/constants/exception.dart:3]:
```suggestion
final failedToReachServerException = Exception('Unable reach the server');
```
do this for all

File: lib/grpc/subscription.dart [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+///
+/// [SubscriptionId] is used for unsubcribing and calling respected stream rpc
+///
+///  Flow
+///  - subscribe to a stream, return [SubscriptionId]
+///  - call the valid server side streaming rpc
+///  - unsubcribe to that stream, with [SubscriptionId]
+///
+class Subscription {
+  final String _sessionid;
```
Comment (Sudhindra3000) [inline lib/grpc/subscription.dart:17]: Probably better if we remove this variable, make the two functions static, and pass sessionId as function parameter

File: lib/global_streams.dart (hunk 1 above)
Comment (siva2204) [reply lib/global_streams.dart:122]: changes as in ? we're catching it in dalalbloc

File: lib/grpc/subscription.dart (hunk 4 above)
Comment (siva2204) [reply lib/grpc/subscription.dart:17]: yeah, that'll be cleaner.

File: lib/grpc/subscription.dart (hunk 2 above)
Comment (siva2204) [reply lib/grpc/subscription.dart:43]: i dont understand this?

caller can do the `error-handling` if there are any errors during subscription.

File: lib/global_streams.dart (hunk 1 above)
Comment (Sudhindra3000) [reply lib/global_streams.dart:122]: I meant why are you catching the exception in this function and throwing the same

Comment (siva2204) [reply lib/global_streams.dart:122]: oh yeah, a bit refactoring is needed in error handling, i added it for other errors that occurs when subscribing

ex: connecting to backend

**Review by Sudhindra3000 — APPROVED** [review]

**Review by satyamurti — APPROVED** [review]

**Pull request merged**